tests/unit_test/channels/whatsapp_test.py

```
@@ -16,6 +17,428 @@
+                args, kwargs = mock_message.call_args
+
+                assert args[0] == bot
+                user_message = args[1]
+                print(user_message.text)
+                assert user_message.text == '/k_multimedia_msg{"latitude": "40.782910059774", "longitude": "-73.959075808525"}'
+
+    @pytest.mark.asyncio
+    async def test_whatsapp_invalid_message_request(self):
+        from kairon.chat.handlers.channels.whatsapp import Whatsapp, WhatsappBot
+        with patch.object(WhatsappBot, "mark_as_read"):
+            with patch.object(Whatsapp, "process_message") as mock_message:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)